### PR TITLE
Resurrecting poll_until

### PR DESCRIPTION
**Pull request by rnirmal**

### CODE REVIEW

**Comment by hub-cap** [comment]
+1

**Comment by sacharya** [comment]
+1

File: reddwarf/common/utils.py [hunk 1]
```diff
@@ -207,6 +208,26 @@ def wait(self):
         return self.done.wait()
 
 
+def poll_until(retriever, condition=lambda value: value,
```
Comment (ed-) [inline reddwarf/common/utils.py:211]: That default lambda is clever and scary at the same time, like all production lambdas I've seen

**Comment by ed-** [comment]
+1

File: reddwarf/common/utils.py (hunk 1 above)
Comment (cp16net) [inline reddwarf/common/utils.py:211]: :mind-blown:

**Comment by cp16net** [comment]
+1

File: reddwarf/common/utils.py (hunk 1 above)
Comment (hub-cap) [inline reddwarf/common/utils.py:211]: craig dont get your mind close to guns dude

**Comment by TimSimpsonR** [comment]
+1

**Pull request merged**